de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/structure/command/RemoveFileStructureCommand.java

```diff
@@ -0,0 +1,114 @@
+
+import java.util.function.Function;
+
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.emf.common.command.AbstractCommand;
+import org.eclipse.core.commands.operations.AbstractOperation;
+
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.model.dvlm.util.command.IVirSatRecordableCommand;
+import de.dlr.sc.virsat.project.Activator;
+import de.dlr.sc.virsat.project.editingDomain.VirSatEditingDomainRegistry;
+import de.dlr.sc.virsat.project.editingDomain.VirSatTransactionalEditingDomain;
+import de.dlr.sc.virsat.project.structure.VirSatProjectCommons;
+import de.dlr.sc.virsat.project.structure.operation.NoUndoDeleteResourceOperation;
+
+/**
+ * This class wraps the IDE Workspace Operation of Deleting the folder of a SEI
+ * By this the deletion is undoable restoring the full content including files
+ * in the SEIs document folder
+ *
+ */
+public class RemoveFileStructureCommand extends AbstractCommand implements IVirSatRecordableCommand {
+
+	public static final Function<IFolder,  ? extends AbstractOperation> DELETE_RESOURCE_OPERATION_FUNCTION
+		= (iFolder) -> new NoUndoDeleteResourceOperation(iFolder, true);
+	
+	private VirSatProjectCommons projectCommons;
+	private AbstractOperation deleteResourceOperation;
+	private VirSatTransactionalEditingDomain editingDomain;
+	
+	/**
+	 * Constructor for creating a command that removes the file structure from a given SEI in the workspace
+	 * @param selectedProject the selected project
+	 * @param iste {@link IStructuralTreeElement}
+	 * @param deleteResourcesOperationFunction the callback function that is used to create the actual delete operation in the workspace
+	 */
+	public RemoveFileStructureCommand(IProject selectedProject, StructuralElementInstance iste, Function<IFolder, ? extends AbstractOperation> deleteResourcesOperationFunction) {
+		projectCommons = new VirSatProjectCommons(selectedProject);
+		IFolder seiFolder = projectCommons.getStructuralElemntInstanceFolder(iste);
+		this.deleteResourceOperation = deleteResourcesOperationFunction.apply(seiFolder);
+		editingDomain = VirSatEditingDomainRegistry.INSTANCE.getEd(iste);
+	}
+	
+	/**
+	 * Wrapper method to exclusively execute a runner in the Virtual Satellite Editing Domain
+	 * @param runnable The runnable to be executed
+	 */
+	protected void executeExclusive(Runnable runnable) {
+		try {
+			editingDomain.runExclusive(runnable);
+		} catch (InterruptedException e) {
+			Activator.getDefault().getLog().log(new Status(Status.ERROR, Activator.getPluginId(), "Failed to Delete SEI folder structure!", e));
+		}
+	}
+	
+	@Override
+	public void execute() {
+		executeExclusive(() -> {
+			try {
+				deleteResourceOperation.execute(null, null);
+				editingDomain.getVirSatCommandStack().triggerSaveAll();

```

Comment:
Could we maybe move the triggerSaveAll() into the executeExclusive(...) method since its logically needed in all cases?

Reply:
Let's give it a try. When i fixed the SWTBot for stability, I realized that these changes easily ended up in deadlocks. Since we solved our deadlock issues, it is worth giving it a try. SWTBot should bail out in case things are bad.